### ISSUE TITLE
Allow waiting on a mock

### ISSUE THREAD

**Issue by pitrou** [python/cpython#61215]
BPO | [17013](https://bugs.python.org/issue17013)
--- | :---
Nosy | @jcea, @pitrou, @rbtcollins, @cjw296, @ezio-melotti, @voidspace, @Kentzo, @lisroach, @mariocj89, @pablogsal, @Kentzo, @tirkarthi
PRs | <li>python/cpython#12818</li><li>python/cpython#16094</li><li>python/cpython#17133</li><li>python/cpython#20759</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2013-01-22.12:15:47.799>
labels = ['type-feature', 'library', '3.9']
title = 'Allow waiting on a mock'
updated_at = <Date 2020-06-10.14:37:15.926>
user = 'https://github.com/pitrou'
```

bugs.python.org fields:
```python
activity = <Date 2020-06-10.14:37:15.926>
actor = 'Kentzo'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2013-01-22.12:15:47.799>
creator = 'pitrou'
dependencies = []
files = []
hgrepos = []
issue_num = 17013
keywords = ['patch']
message_count = 18.0
messages = ['180377', '181398', '247777', '340146', '340153', '340158', '340164', '352211', '356898', '367446', '371133', '371165', '371175', '371193', '371195', '371196', '371202', '371207']
nosy_count = 13.0
nosy_names = ['jcea', 'pitrou', 'rbcollins', 'cjw296', 'ezio.melotti', 'michael.foord', 'Ilya.Kulakov', 'lisroach', 'mariocj89', 'lkollar', 'pablogsal', 'Kentzo', 'xtreak']
pr_nums = ['12818', '16094', '17133', '20759']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue17013'
versions = ['Python 3.9']
```

</p></details>

### Linked PRs
* gh-106366
* gh-106414
* gh-106591
* gh-106822

**Comment by terryjreedy** [comment]
The 5 attached PRs seem to have already gone beyond the original issue.  Let's open a new issue for anything else.
